lib/utils/background_service.dart

```diff
@@ -0,0 +1,126 @@
+class BackgroundService {
+  static Future<DriftIsolate?> get _isolate async {
+    final token = RootIsolateToken.instance;
+    if (token == null) {
+      SimpleLogger().warning("RootIsolateToken is null.");
+      return null;
+    }
+    return await DriftIsolate.spawn(() {
+      BackgroundIsolateBinaryMessenger.ensureInitialized(token);
+
+      return LazyDatabase(() async {
+        final dbFolder = await getApplicationDocumentsDirectory();
+        final file = File(p.join(dbFolder.path, "db.sqlite"));
+
+        return NativeDatabase(file);
+      });
+    });
+  }
+
+  /// Callback for messages sent from the isolate
+  ///
+  /// This is used to catch [SurveyAnswerException]s and report them with Sentry
+  static Future<void> _onIsolateSurveyAnswerException(dynamic message) async {
+    if (message is SurveyAnswerException) {
+      await reportError(message, null);
+    }
+  }
+
+  /// Starts the background service
+  static Future<void> start(String apiBasePath) async {
+    SimpleLogger().info("Starting background service...");
+    final receivePort = ReceivePort();
+
+    receivePort.listen(_onIsolateSurveyAnswerException);
+    DriftIsolate? driftIsolate = await _isolate;
+    if (driftIsolate == null) {
+      SimpleLogger()
+          .warning("DriftIsolate is null. Aborting background service.");
+      return;
+    }
+    Isolate.spawn(
+      _startIsolate,
+      BackgroundServiceArgs(apiBasePath, receivePort.sendPort, driftIsolate),
+    );
+  }
+
+  /// The entry point for the isolate
+  ///
+  /// It opens a background connection to the database and every 30 minutes checks for unsent answers in batches of 10.
+  /// If unsent answers are found, it attempts to send them to the backend.
+  /// If it success, it deletes the answer from the local database.
+  /// If it fails, it sends the exception back to the main isolate.
+  static Future<void> _startIsolate(BackgroundServiceArgs args) async {
+    final conn = await args.driftIsolate.connect();
+    final db = Database.fromQueryExecutor(conn);
+    final answerDao = AnswersDao(db);
+    final keysDao = KeysDao(db);
+    Timer.periodic(const Duration(minutes: 30), (_) async {
+      SimpleLogger().info("Checking for unsent answers");
+      final unsentAnswers = await answerDao.listAnswers(limit: 10);
+      final deviceKey = await keysDao.getDeviceKey();
+      final deviceId = await keysDao.getDeviceId();
+      final api = await ApiFactory().getDeviceDataApi(
+          overrideApiBasePath: args.apiBasePath, overrideDeviceKey: deviceKey);
+      if (deviceId == null) {
+        SimpleLogger().warning("Device ID is null.");
+        return;
+      }
+
+      SimpleLogger().info("Unsent answers: ${unsentAnswers.length}");
+      for (final answer in unsentAnswers) {
+        try {
+          SimpleLogger().info("Attempting to send unsent answer: $answer");
+          final builtAnswer = DevicePageSurveyAnswer((builder) {
```

Comment:
We need to include the original submission time for the request

Reply:
API doesn't yet support that, I'll include that in another PR.